src/PJ_helmert.c

```diff
@@ -520,12 +520,15 @@ PJ *PROJECTION(helmert) {
     if (pj_param (P->ctx, P->params, "ttheta").i) {
         Q->theta_0 = pj_param (P->ctx, P->params, "dtheta").f * ARCSEC_TO_RAD;
         Q->fourparam = 1;
+        Q->scale_0 = 1.0; /* default scale for the 4-param shift */
     }
 
     /* Scale */
-    if (pj_param (P->ctx, P->params, "ts").i)
+    if (pj_param (P->ctx, P->params, "ts").i) {
         Q->scale_0 = pj_param (P->ctx, P->params, "ds").f;
-
+        if (pj_param (P->ctx, P->params, "ttheta").i && Q->scale_0 == 0.0)
+            return freeup_msg(P, -52);
```

Comment:
Converting everything might be good indeed but tedious... But I see someone (likely @busstoptaktik if it is not you) added a few PJD_ERR_ constants in projects.h, so I think we should at least use that new good habit for new codes.